### PR TITLE
Add a feature for fetching renames (across SHAs) from GitHub

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
See https://github.com/web-platform-tests/wpt.fyi/issues/663

Unfortunately, go-github doesn't have the `previous_filename` field, so we have a bit more boilerplate than I'd like.

## Review Information
Canonical example, `shape-outside-inset-010.html` renamed to `shape-outside-inset-0010.html`:
https://renames-dot-wptdashboard-staging.appspot.com/results/css/css-shapes/shape-outside/values?products=chrome%5Bexperimental%5D@4b370b4c0b,chrome%5Bexperimental%5D@bb9c0ce15e&diff=true&q=shape-outside-inset-0

### CODE REVIEW

**Comment by foolip** [comment]
I haven't reviewed yet, but in https://github.com/web-platform-tests/wpt/pull/13971 I create a commit that changes a file that's been modified on master, and if merged (using `git merge` at least) would be conflict free and just result in the changes being applied to another file.

Can you try this commit with the relevant GitHub APIs to see if it claims there are renames? If a GitHub status/checks run can start at all, it would see the post-rename filename. Not sure if any confusion is possible from this.

**Comment by lukebjerring** [comment]
I'm not sure what you're speculating here. Essentially this feature is "ask GitHub what got renamed", so we're bound by their implementation either way.

**Comment by foolip** [comment]
I think https://api.github.com/repos/web-platform-tests/wpt/compare/master...3eb426c4096455b1f4a2f9d0cb18561c183864d0 is the API call that would result for https://github.com/web-platform-tests/wpt/pull/13971. Response is:
```
  "files": [
    {
      "sha": "295441728233ffdf9b1ef17d55706dca3ca271c6",
      "filename": "content-security-policy/navigate-to/form-action/form-action-allows-navigate-to-allows.html",
      "status": "modified",
      "additions": 0,
      "deletions": 2,
      "changes": 2,
      "blob_url": "https://github.com/web-platform-tests/wpt/blob/3eb426c4096455b1f4a2f9d0cb18561c183864d0/content-security-policy/navigate-to/form-action/form-action-allows-navigate-to-allows.html",
      "raw_url": "https://github.com/web-platform-tests/wpt/raw/3eb426c4096455b1f4a2f9d0cb18561c183864d0/content-security-policy/navigate-to/form-action/form-action-allows-navigate-to-allows.html",
      "contents_url": "https://api.github.com/repos/web-platform-tests/wpt/contents/content-security-policy/navigate-to/form-action/form-action-allows-navigate-to-allows.html?ref=3eb426c4096455b1f4a2f9d0cb18561c183864d0",
      "patch": "@@ -1,10 +1,8 @@\n <!DOCTYPE html>\n-\n <head>\n <script src=\"/resources/testharness.js\"></script>\n <script src=\"/resources/testharnessreport.js\"></script>\n </head>\n-\n <body>\n <script>\n   var t = async_test(\"Test that form-action overrides navigate-to when present.\");"
    }
  ]
```

form-action-allows-navigate-to-allows.html was the name before renaming, no rename detected there, as expected actually since `A...B` compares the merge base of A and B to B.

However, it might actually not matter. If we've tested some version of master after the rename and the PR merged into that same post-rename commit, then both runs will have seen the rename.

File: api/diff.go [hunk 1]
```diff
@@ -154,11 +171,83 @@ func handleAPIDiffPost(w http.ResponseWriter, r *http.Request) {
...
+	diffJSON := shared.GetResultsDiff(beforeJSON, afterJSON, filter, paths, nil)
 	var bytes []byte
 	if bytes, err = json.Marshal(diffJSON); err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 	w.Write(bytes)
 }
+
+// Wrappers for including a missing field in go-github
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply api/diff.go]: https://github.com/google/go-github/pull/1043 has landed. Will clean this up.

**Pull request merged**